internal/xpkg/fetch.go

```diff
@@ -136,7 +136,15 @@ func (i *K8sFetcher) Head(ctx context.Context, ref name.Reference, secrets ...st
 	if err != nil {
 		return nil, err
 	}
-	return remote.Head(ref, remote.WithAuthFromKeychain(auth), remote.WithTransport(i.transport), remote.WithContext(ctx))
+	d, err := remote.Head(ref, remote.WithAuthFromKeychain(auth), remote.WithTransport(i.transport), remote.WithContext(ctx))
+	if err != nil || d == nil {
+		rd, gErr := remote.Get(ref, remote.WithAuthFromKeychain(auth), remote.WithTransport(i.transport), remote.WithContext(ctx))
+		if gErr != nil {
+			return nil, errors.Wrapf(gErr, "Failed to fetch package descriptor with a GET request after a previous HEAD request failure: %v", err)
```

Comment:
Was initially wondering if `remote.Head` was returning an error on successful responses that just didn't include the digest header, but it looks like it is, so this error message should always have a valid `err`. 👍🏻 [ref](https://github.com/google/go-containerregistry/blob/87b3a7966622dbfdb8cbb9c4b5068884540d387e/pkg/v1/remote/descriptor.go#L340)

Reply:
```suggestion
			return nil, errors.Wrapf(gErr, "failed to fetch package descriptor with a GET request after a previous HEAD request failure: %v", err)
```

ref: https://github.com/golang/go/wiki/CodeReviewComments#error-strings